### PR TITLE
Accelerate multi-add for intel64 backend

### PR DESCRIPTION
**Pull request by LuoYuanke**
ideep backend has the ability to fuse several add operation into one sum primitive. This PR is to accelerate the mutli_add on intel64 backend.

### CODE REVIEW

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Could you please add test code?
(i.e., fix `TestMultipleAdd` in `chainer_tests/functions_tests/math_tests/test_basic_math.py` to use `backend.inject_backend_tests`)
You can see the usage of `inject_backend_tests` in iDeep PRs like #4277.

**Comment by LuoYuanke** [comment]
Thanks for the instruction. The unit test is for backend is added.

**Comment by LuoYuanke** [comment]
@kmaehashi 
Could you help to review?

File: tests/chainer_tests/functions_tests/math_tests/test_basic_math.py [hunk 1]
```diff
@@ -245,41 +258,45 @@ def setUp(self):
         self.ggx2 = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
         self.ggx3 = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
 
-    def check_forward(self, func, x1_data, x2_data, x3_data):
+    def check_forward(self, func, x1_data, x2_data, x3_data, backend_config):
```
Comment (kmaehashi) [inline tests/chainer_tests/functions_tests/math_tests/test_basic_math.py]: You need to manually transfer all inputs to GPU if `backend_config.use_cuda` is `True`.
https://github.com/chainer/chainer/pull/4277/files?diff=split#diff-50ce841c2d595fcd3d6e74f9c363c260R60
Same for `check_backward` and `check_double_backward`.

**Comment by kmaehashi** [comment]
Thanks for the fix!
Please check coding style violations: https://travis-ci.org/chainer/chainer/jobs/352192823#L1118
LGTM otherwise.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Are you finished working on this PR? Can I proceed to merge?

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by LuoYuanke** [comment]
I think I'm finished on this PR, but I encounter some problem on pooling test case. I think it is precision issue, so I have to re-trigger test for several times. If you can ignore this problem, please merge the code. Thanks.

**Comment by kmaehashi** [comment]
Thanks. The failure seems unrelated to your change.
(you don't have to modify code just to retrigger CI; we'll do that if we think we need.)

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**